### ISSUE TITLE
[10.0] UnicodeEncodeError if partner email contains non-ascii characters

### ISSUE THREAD

**Issue by aisopuro** [odoo/odoo#19852]
#### Impacted versions:
10.0. Tested on http://269067-10-0-1aa8de.runbot8.odoo.com, which was latest at time of writing.

#### Steps to reproduce:
1. Start creating a new partner
2. Enter an email-address that contains non-ascii characters (I used `Lärry@gmail.com`)
3. Save

#### Current behavior:
An error popup is shown with a traceback (traceback is at the bottom of this description)

#### Expected behavior:
I see three options:
- Odoo should properly allow email addresses to contain non-ascii characters (this has been the standard since 2015: https://tools.ietf.org/html/rfc6532)
- There should be a clear warning that states that Odoo does not support non-ascii characters in email addresses.
- Fetching the Gravatar image should fail silently if it gets an encodingerror, though this I think is only acceptable if Gravatar itself does not support non-ascii email addresses. 

I would obviously prefer the first, but I don't really know if it's a harder problem than it seems to be.

#### Traceback:

```
Traceback (most recent call last):
  File "/home/odoo/runbot/extra/runbot/static/build/269067-10-0-1aa8de/odoo/http.py", line 640, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/home/odoo/runbot/extra/runbot/static/build/269067-10-0-1aa8de/odoo/http.py", line 677, in dispatch
    result = self._call_function(**self.params)
  File "/home/odoo/runbot/extra/runbot/static/build/269067-10-0-1aa8de/odoo/http.py", line 333, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/home/odoo/runbot/extra/runbot/static/build/269067-10-0-1aa8de/odoo/service/model.py", line 101, in wrapper
    return f(dbname, *args, **kwargs)
  File "/home/odoo/runbot/extra/runbot/static/build/269067-10-0-1aa8de/odoo/http.py", line 326, in checked_call
    result = self.endpoint(*a, **kw)
  File "/home/odoo/runbot/extra/runbot/static/build/269067-10-0-1aa8de/odoo/http.py", line 935, in __call__
    return self.method(*args, **kw)
  File "/home/odoo/runbot/extra/runbot/static/build/269067-10-0-1aa8de/odoo/http.py", line 506, in response_wrap
    response = f(*args, **kw)
  File "/home/odoo/runbot/extra/runbot/static/build/269067-10-0-1aa8de/odoo/addons/web/controllers/main.py", line 885, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/home/odoo/runbot/extra/runbot/static/build/269067-10-0-1aa8de/odoo/addons/web/controllers/main.py", line 877, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/home/odoo/runbot/extra/runbot/static/build/269067-10-0-1aa8de/odoo/api.py", line 689, in call_kw
    return call_kw_multi(method, model, args, kwargs)
  File "/home/odoo/runbot/extra/runbot/static/build/269067-10-0-1aa8de/odoo/api.py", line 680, in call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/home/odoo/runbot/extra/runbot/static/build/269067-10-0-1aa8de/odoo/models.py", line 5496, in onchange
    record._onchange_eval(name, field_onchange[name], result)
  File "/home/odoo/runbot/extra/runbot/static/build/269067-10-0-1aa8de/odoo/models.py", line 5394, in _onchange_eval
    method_res = method(self)
  File "/home/odoo/runbot/extra/runbot/static/build/269067-10-0-1aa8de/odoo/addons/base/res/res_partner.py", line 363, in onchange_email
    self.image = self._get_gravatar_image(self.email)
  File "/home/odoo/runbot/extra/runbot/static/build/269067-10-0-1aa8de/odoo/addons/base/res/res_partner.py", line 699, in _get_gravatar_image
    email_hash = hashlib.md5(email.lower()).hexdigest()
UnicodeEncodeError: 'ascii' codec can't encode character u'\xe4' in position 1: ordinal not in range(128)
```

**Comment by aisopuro** [comment]
The code that produces the error was originally committed by @dbu-odoo and @tde-banana-odoo in 66d4de7a0f99555e8d13fcd52d7fe4d3b92cc9fd. Maybe one of them could take a look?

**Comment by Yenthe666** [comment]
Hi @aisopuro,

This is indeed a bug:
![image](https://user-images.githubusercontent.com/6352350/31117317-b0b30978-a829-11e7-928b-e4b70ca3c22e.png)
Encoding needs to be added.

Regards,
Yenthe

**Comment by Yenthe666** [comment]
I made an official ticket for this one - 1837714.
This has been fixed in 11.0 in the meanwhile though!

**Comment by Yenthe666** [comment]
Fixed with https://github.com/odoo/odoo/commit/e4731d0327b77da7f3013c13caa606951dbb0a69 in 10.0!
